doc/api/fs.md

```diff
@@ -1395,6 +1403,224 @@ The synchronous version of [`fs.writeFile()`][]. Returns `undefined`.
+link.</td>
+</tr>
+<tr>
+<td><code>O_SYNC</code></td>
+<td>Flag indicating that the file is opened for synchronous I/O.</td>
+</tr>
+<tr>
+<td><code>O_SYMLINK</code></td>
+<td>Flag indicating to open the symbolic link itself rather than the resource it
+is pointing to.</td>
+</tr>
+<tr>
+<td><code>O_DIRECT</code></td>
+<td>When set, an attempt will be made to minimize caching effects of file
+I/O.</td>
+</tr>
+<tr>
+<td><code>O_NONBLOCK</code></td>
+<td>Flag indicating to open the file in nonblocking mode when possible.</td>
+</tr>
+</table>
+
+### File Type Constants
+
+The following constants are meant for use with the [`fs.Stats`][] object's
+`mode` property for determining a file's type.
+
+<table>
+<tr>
+<th>Constant</th>
+<th>Description</th>
+</tr>
+<tr>
+<td><code>S_IFMT</code></td>
+<td>Bit mask used to extract the file type code.</td>
+</tr>
+<tr>
+<td><code>S_IFREG</code></td>
+<td>File type constant for a regular file.</td>
+</tr>
+<tr>
+<td><code>S_IFDIR</code></td>
+<td>File type constant for a directory.</td>
+</tr>
+<tr>
+<td><code>S_IFCHR</code></td>
+<td>File type constant for a character-oriented device file.</td>
+</tr>
+<tr>
+<td><code>S_IFBLK</code></td>
+<td>File type constant for a block-oriented device file.</td>
+</tr>
+<tr>
+<td><code>S_IFIFO</code></td>
+<td>File type constant for a FIFO/pipe.</td>
+</tr>
+<tr>
+<td><code>S_IFLNK</code></td>
+<td>File type constant for a symbolic link.</td>
+</tr>
+<tr>
+<td><code>S_IFSOCK</code></td>
+<td>File type constant for a socket.</td>
+</tr>
+</table>
+
+### File Mode Constants
+
+The following constants are meant for use with the [`fs.Stats`][] object's
+`mode` property for determining the access permissions for a file.
+
+<table>
+<tr>
+<th>Constant</th>
+<th>Description</th>
+</tr>
+<tr>
+<td><code>S_IRWXU</code></td>
+<td>File mode indicating read, write and execute by owner.</td>
+</tr>
+<tr>
+<td><code>S_IRUSR</code></td>
+<td>File mode indicating read by owner.</td>
+</tr>
+<tr>
+<td><code>S_IWUSR</code></td>
+<td>File mode indicating write by owner.</td>
+</tr>
+<tr>
+<td><code>S_IXUSR</code></td>
+<td>File mode indicating executable by owner.</td>
+</tr>
+<tr>
+<td><code>S_IRWXG</code></td>
+<td>File mode indicating read, write and execute by group.</td>
+</tr>
+<tr>
+<td><code>S_IRGRP</code></td>
+<td>File mode indicating read by group.</td>
+</tr>
+<tr>
+<td><code>S_IWGRP</code></td>
+<td>File mode indicating write by group.</td>
+</tr>
+<tr>
+<td><code>S_IXGRP</code></td>
+<td>File mode indicating executable by group.</td>
+</tr>
+<tr>
+<td><code>S_IRWXO</code></td>
+<td>File mode indicating read, write and execute by others.</td>
+</tr>
+<tr>
+<td><code>S_IROTH</code></td>
+<td>File mode indicating read by others.</td>
+</tr>
+<tr>
+<td><code>S_IWOTH</code></td>
+<td>File mode indicating write by others.</td>
+</tr>
+<tr>
+<td><code>S_IXOTH</code></td>
+<td>File mode indicating executable by others.</td>
```

Comment:
nit: would use “readable”/“writable”/“executable” in all of these descriptions for consistency/clarity